history/metricq-history.js

```diff
@@ -108,6 +108,52 @@ class AnalyzeQuery {
   }
 }
 
+class HtaQuery{
+  constructor (mq, from, to, points) {
+    this.mq = mq
+    this.from = from
+    this.to = to
+    this.points = points
+    this.metrics = []
+  }
+
+  metric (metric) {
+    this.metrics.push(
+      metric
+    )
+
+    return this
+  }
+
+  _parse_result (result) {
+    let data = result['data']
+    for(const metric_data of Object.values(data)) {
+      for (const value of metric_data[metric_data.mode]) {
```

Comment:
```suggestion
      for (const value of metric_data[metric_data.mode]) {
```

How does this even work? `value` is modified.

Reply:
Only value.time gets modified. const only stops reassigning value itself.